packages/react-magma-dom/src/components/StyledButton/styles.ts

```diff
@@ -134,53 +166,75 @@ export function buildBorderColor(props) {
 export function buildColor(props) {
   props = buildPropsWithDefaultButtonStyles(props);
 
-  if (props.disabled && props.isInverse && props.variant !== 'solid') {
-    return props.theme.colors.disabledInverseText;
+  if (props.variant === 'link' && !props.isInverse && !props.disabled) {
+    if (props.color === 'danger') {
+      return props.theme.colors.danger500;
+    }
+    return props.theme.colors.primary;
   }
+
   if (props.disabled) {
-    return props.theme.colors.disabledText;
-  }
-  if (props.color === 'marketing') {
-    return props.theme.colors.foundation02;
+    if (props.isInverse) {
+      if (props.color === 'secondary') {
+        return transparentize(0.3, props.theme.colors.neutral100);
+      }
+      return transparentize(0.2, props.theme.colors.neutral100);
+    }
+    return transparentize(0.4, props.theme.colors.neutral500);
   }
-  if (
-    (!props.isInverse && props.variant === 'solid') ||
-    (props.isInverse && props.variant !== 'solid')
-  ) {
-    if (props.color === 'secondary' && !props.isInverse) {
-      return props.theme.colors.neutral;
+
+  if (props.isInverse) {
+    if (props.variant === 'link') {
+      if (props.color === 'danger') {
+        return props.theme.colors.danger200;
+      }
+      return props.theme.colors.tertiary500;
+    }
+    if (props.color === 'secondary') {
+      return props.theme.colors.tertiary500;
+    }
+    if (props.color === 'marketing') {
+      return props.theme.colors.primary500;
+    }
+    if (props.color === 'danger') {
+      return props.theme.colors.neutral100;
     }
-    return props.theme.colors.neutral08;
+    return props.theme.colors.primary700;
   }
+
   switch (props.color) {
     case 'secondary':
-      return props.theme.colors.neutral;
-    case 'success':
-      return props.theme.colors.success;
-    case 'danger':
-      return props.theme.colors.danger;
-    default:
       return props.theme.colors.primary;
+    case 'marketing':
+      return props.theme.colors.primary;
+    default:
+      return props.theme.colors.neutral100;
```

Comment:
I'm probably missing it, but do we have the check for `danger` when not inverse and not disabled, or do we expect it to be this default?

Reply:
Looks like it expects to be white, so it defaults which is fine.